packages/gitbook/src/components/Search/SearchInput.tsx

```diff
@@ -0,0 +1,148 @@
+                        'theme-bold:has-[input:focus-visible]:bg-header-link/3 theme-bold:has-[input:focus-visible]:ring-header-link/6',
+                        'theme-bold:before:absolute theme-bold:before:inset-0 theme-bold:before:bg-header-background/7 theme-bold:before:backdrop-blur-xl ', // Special overlay to make the transparent colors of theme-bold visible.
+                        'relative z-30 max-md:absolute max-md:right-0',
+                        className
+                    )}
+                >
+                    {value ? (
+                        <Button
+                            variant="blank"
+                            label="Clear"
+                            size="medium"
+                            iconOnly
+                            icon="circle-xmark"
+                            className="-mx-1.5 animate-scaleIn px-1.5 theme-bold:text-header-link theme-bold:hover:bg-header-link/3"
+                            onClick={() => {
+                                onChange({
+                                    target: { value: '' },
+                                } as React.ChangeEvent<HTMLInputElement>);
+                                inputRef.current?.focus();
+                            }}
+                        />
+                    ) : (
+                        <Icon icon="magnifying-glass" className="size-4 animate-scaleIn" />
+                    )}
+
+                    <input
+                        type="text"
+                        onFocus={onFocus}
+                        onKeyDown={onKeyDown}
+                        onChange={onChange}
+                        value={value ?? ''}
+                        size={1} // Determines the width of the input (in characters). It's inconsistent between browsers and limits the min-width, so we set it to a sensible minimum and control width ourselves.
+                        placeholder={`${tString(language, withAsk ? 'search_or_ask' : 'search')}...`}
+                        className={tcls(
+                            'peer z-10 grow bg-transparent py-0.5 text-tint-strong theme-bold:text-header-link outline-none transition-[width] duration-300 placeholder:text-tint theme-bold:placeholder:text-current theme-bold:placeholder:opacity-7',
+                            value !== undefined
+                                ? 'max-md:w-40'
+                                : 'max-md:-ml-2 max-md:w-0 max-md:opacity-0'
+                        )}
+                        ref={inputRef}
+                    />
+                    <Shortcut />
+                </div>
+            </div>
+        );
+    }
+);
+
+function Shortcut() {
+    const [operatingSystem, setOperatingSystem] = useState<string | null>(null);
+
+    useEffect(() => {
+        function getOperatingSystem() {
```

Comment:
Why do you have this function here? if it's global define it outside the component

Otherwise it's often redefined by the JS engine